pkg/frontend/frontend.go

```diff
@@ -252,6 +237,24 @@ func (f *frontend) Run(ctx context.Context, stop <-chan struct{}, done chan<- st
 	}
 }
 
+func (f *frontend) Shutdown(done chan<- struct{}) {
```

Comment:
Not sure if at this point it's worth introducing the same function for the backend for consistency as logically it is not needed.